algorand.cpp

```diff
@@ -232,21 +241,89 @@ msgpack::packer<Stream>& Subsig::pack(msgpack::packer<Stream>& o) const {
 }
 
 MultiSig::MultiSig(std::vector<Address> addrs, uint64_t threshold) :
-  threshold(threshold ? threshold : addrs.size()) {
+  sigs{},     
+  threshold{threshold ? threshold : addrs.size()},
+  public_address{} {
   for (const auto& addr : addrs) {
-    sigs.push_back(Subsig(addr.public_key));
+     sigs.push_back(Subsig(addr.public_key));
   }
+  this->update_address();
 }
 
+//Note: msig address is part of the transaction and gets packed
+//as part of the transaction
 template <typename Stream>
 msgpack::packer<Stream>& MultiSig::pack(msgpack::packer<Stream>& o) const {
   o.pack_map(3);
-  o.pack("subsigs"); o.pack(sigs);
+  o.pack("subsig");
```

Comment:
Does kv_pack not work here, as on lines below? I do it that way, even though it is not much savings, so that one can easily verify (and correct) mistakes in ordering by simply sorting the lines lexicographically (as our msgpack demands)

Reply:
Yes,  ```kv_pack(o, "subsig", sigs)``` should work just as well. I wanted to leave it the way you had it since the o.pack() is overloaded for vector<SubSig> anyways and worked as well. 

I'll update this section to use kv_pack so it's more consistent.

Reply:
I wonder why I had it that way.  Maybe I got stuck in the middle of something.